### ISSUE TITLE
Add support for MCP3421

### ISSUE THREAD

**Issue by dotpointer** [stevemarple/MCP342x#11]
MCP3421 is 18-bit but only 1 channel.

I succeeded to use your library with a MCP3421 module by commenting out all channels but 1 in the example, here are the relevant lines that I needed to change:

`addr68_ch0 = MCP342x(bus, 0x68, channel=0, resolution=18)`
...
`adcs = [addr68_ch0]`

With these changes I could read data from the module. So you can add support for the board, it works. :smiley:

**Comment by ouccoffee** [comment]
I think you should talk about it in python-MCP342x， but not here.

**Comment by Andy79881** [comment]
Can I use your arduino library for MCP3421A0 device and if yes how to setup correctly ?

**Comment by stevemarple** [comment]
> Can I use your arduino library for MCP3421A0 device and if yes how to setup correctly ?

Yes, you need to make sure you use the correct address (0x68 for the A0 part):

~~~ C++
MCP342x adc = MCP342x(0x68);
~~~

Only use `channel1` with that part, as the other channels are not present.

**Comment by Andy79881** [comment]
Thanks for setup hints. I've got the code running.
I am using Spence Kondes AVR Dx core for AVR32DD20 device. Everything works, but the SW stopps at executing from your ConvertAndRead.ino example:

```
 uint8_t err = adc.convertAndRead(MCP342x::channel1, MCP342x::oneShot,
				   MCP342x::resolution16, MCP342x::gain1,
				   1000000, value, status);
```

It stopps there, because the code following this line is not executed. I have commented this line and then my remaining code worked as expected. This may worked for Atmega328, I have not tried since I used AVR DD series device from the begining.

Now I'm using the routines used in your ConvertAndReadNoDelay.ino example (but with delay ...):

```
err = adc.convert(config); 
_delay_ms(55); // works with 50ms, set to 55ms
err = adc.read(value, status);

```
As you can see I do not use the flag method, simple a (blocking) delay. With this code everything works but I will try to change the code using the flag in order to avoid the long delay of ~55ms.

Can you confirm, that the status.isReady() == TRUE if conversion is finished?

**Comment by stevemarple** [comment]
Everything works for MCP3424. It is mature code and I have been running it in production systems for over 10 years. If you want to use `isReady()` you should do so like this:

~~~ C++
if (!err && status.isReady()) {
// Have valid data
...
}
~~~

I haven't used the MCP3421 but having checked the datasheet it looks completely compatible with MCP3422/MCP3423/MCP3424 as long as you only use `channel1`.

In your case I think the unknown is using the AVR32 core. Are you able to test using a standard Arduino device?

**Comment by Andy79881** [comment]
No, I have not the possibility to test it now but I guess your code will work fine on original Arduino. I have simply ported an existing design to AVR32DD20 device.
